### PR TITLE
Upgraded to Pester v5.3.0

### PR DESCRIPTION
**Pull request by ArmaanMcleod**
## PR Summary

Upgraded to latest Pester 5.3.0 version which is stable.

https://github.com/pester/Pester/releases/tag/5.3.0

Had to make some tweaks to module scoping in `Cmdlet.Common.Tests.ps1` but other than that quite a straightforward upgrade. The benefit of upgrading is now CI errors are red in the build log, and has a bunch of handy new features in that release I added 😄 . 

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [ ] Unit tests created/ updated
  - [ ] Rule documentation created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [x] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by ArmaanMcleod** [comment]
Might hold off on this for now. I realised that there might be a slight bug in the `Output.CIFormat` option I added to Pester 5.3.0 where error logs show up as red in the CI. Will follow up with Pester maintainer to see if this is something I should fix. Doesn't really have any impact on this project, given I made the upgrade work with some tweaks to module scoping(no idea why this broke between 5.2.2 and 5.3.0).

Can keep this here if we need 5.3.0 features in the future. Will probably be fine sticking with 5.2.2 for now.

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod 👍  All good to merge.

**Pull request merged**
